### DISCUSSION TITLE
Using Grafana Plugins in an HA Configuration

### DISCUSSION THREAD

**Discussion by mari-arondeus** [grafana/grafana#56586] [Plugins]
Hello, all!

We're currently moving our Grafana cluster to an HA configuration, and the main sticking point appears to be plugins. Our SQL backend is working great for dashboards and configuration, but since the Grafana instances roam between host nodes they aren't able to keep the same plugins. We could add a local volume on each host with the plugin files, but this would prevent us from performing automatic upgrades. Another option would be to create a PV, but this would have similar issues - we're also trying to avoid creating a PV for this purpose.

I'm curious as to how others are handling this - are you mounting a PV or an S3 bucket for your plugin files? Has that worked well for you, or are there related issues?

Any input on this would be fantastic. Thank you!

**Comment by mari-arondeus** [comment]
"Note: If you need to specify the version of a plugin, then you can add it to the GF_INSTALL_PLUGINS environment variable. Otherwise, the latest is used. For example: -e "GF_INSTALL_PLUGINS=grafana-clock-panel 1.0.1,grafana-simple-json-datasource 1.3.5"."

Source: [https://grafana.com/docs/grafana/latest/setup-grafana/installation/docker/](https://grafana.com/docs/grafana/latest/setup-grafana/installation/docker/)

Welp, that's a big RTFM moment. Sorry!

**Comment by tolzhabayev** [comment]
So I assume this discussion can be closed?

**Reply by mari-arondeus** [reply]
Yes, please feel free!
